src/expression/evaluation_context.ts

```diff
@@ -29,8 +39,59 @@ class EvaluationContext {
         return this.feature && 'id' in this.feature ? this.feature.id : null;
     }
 
+    // Bluntly copied from
+    // https://github.com/mapbox/vector-tile-js/blob/77851380b63b07fd0af3d5a3f144cc86fb39fdd1/lib/vectortilefeature.js#L225-L233
+    _signedArea(ring) {
```

Comment:
Is it possible that some of the methods added here already exists in the following file: 
https://github.com/maplibre/maplibre-style-spec/blob/63b660a03a444c358d8d363d578e8bf29e2a8c3d/src/util/classify_rings.ts#L56

Reply:
Done in [cbdd941](https://github.com/maplibre/maplibre-style-spec/pull/519/commits/cbdd9418c3c56469e56d297e15f6be946bbe5511)